### PR TITLE
LOGGING & ERROR HANDLING

### PR DESCRIPTION
**Pull request by diabolicsound**
Implement hometask 5

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -9,7 +11,17 @@ app.use(express.json());
...
+app.use('/', methodErrorHandler);
 
 app.listen(port, () => {
-    console.log(`Server is running at http://localhost:${port}`);
+    logger.info(`Server is running at http://localhost:${port}`);
 });
+
+process
+    .on('unhandledRejection', (reason, p) => {
+        logger.error('Unhandled Rejection at Promise', { reason, promise: p });
```
Comment (ilya-evseev) [inline src/index.js:22]: Maybe it's good to exit(1) on rejection too

File: src/users/controllers/users.controller.js [hunk 2]
```diff
@@ -1,30 +1,76 @@
...
     addUser: async (req, res) => {
+        const { params } = req;
+        infoMiddleware(req);
+
         const newUser = req.body;
-        res.json(await usersList.createUser(newUser));
+        try {
+            res.json(await usersList.createUser(newUser));
+        } catch (e) {
+            errorMiddleware(e, userMethods.getUsers, params, res, true);
```
Comment (ilya-evseev) [inline src/users/controllers/users.controller.js:27]: userMethods.addUser?

File: src/users/controllers/users.controller.js [hunk 3]
```diff
@@ -1,30 +1,76 @@
...
         const updates = req.body;
-        const { id } = req.params;
-        res.json(await usersList.updateUser(id, updates));
+        const { id } = params;
+        infoMiddleware(req);
+
+        try {
+            res.json(await usersList.updateUser(id, updates));
+        } catch (e) {
+            errorMiddleware(e, userMethods.getUsers, params, res, true);
```
Comment (ilya-evseev) [inline src/users/controllers/users.controller.js:51]: wrong method name?

File: src/users/controllers/users.controller.js [hunk 4]
```diff
@@ -1,30 +1,76 @@
...
+import { infoMiddleware } from '../../log-middlewares/info.log.middleware.js';
+import { errorMiddleware } from '../../log-middlewares/error.log.middleware.js';
+
+import { userMethods } from '../constants/users.constants.js';
 
 export const controllers = {
     getUsers: async (req, res) => {
-        res.json(await usersList.getUsers());
+        const { params } = req;
+        infoMiddleware(req);
```
Comment (ilya-evseev) [inline src/users/controllers/users.controller.js:10]: In general, it is not a middleware. It's just a method. A middleware is used like app.use(yourMiddleware) and has a special signature. So, I think, it's good to rename

File: src/users/controllers/users.controller.js [hunk 5]
```diff
@@ -1,30 +1,76 @@
...
+
+        try {
+            res.json(await usersList.getUsers());
+        } catch (e) {
+            errorMiddleware(e, userMethods.getUsers, params, res);
```
Comment (ilya-evseev) [inline src/users/controllers/users.controller.js:15]: In general, it is not a middleware. It's just a method. A middleware is used like app.use(yourMiddleware) and has a special signature. So, I think, it's good to rename

**Review by ilya-evseev — COMMENTED** [review]
The 1st point in task 5.2 is not implemented - please, add APP-level middleware to log all unhandled errors and return 500

**Comment by diabolicsound** [comment]
Updated upon listed notes.

**Pull request merged**